### PR TITLE
push to the public repository

### PR DESCRIPTION
**Pull request by vegegoku**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    ```typescript
+    export class TypeSimple {
+        name:string;
+    
+        constructor();
+    }
+    ```
+- `@TsTypeRef` : Use this annotation on a type, parameter, method, or field to override the element type by the type of another element.
+
+    **Example** :
```
Comment (niloc132) [inline README.md:63]: I'm not sure about these examples - do we think we often want to emit TS to global when the JS is in a namespace?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+**/*/node
+**/*/node_modules
+**/*/package.json
+**/*/package-lock.json
+**/*/target
```
Comment (niloc132) [inline .gitignore:5]: Can't we just put `target/` etc here to declare files that might be at any path? I know for example that pom.xml.versionsBackup could be at any path, but that doesnt have the wildcard prefix

File: jsinterop-ts-defs-processor/pom.xml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                        <configuration>
+                            <!-- optional: The default argument is actually
+                            "install", so unless you need to run some other npm command,
+                            you can remove this whole <configuration> section.
+                            -->
+                            <arguments>check-dts **/target/generated-test-sources/test-annotations/**/tsinterop.*.ts</arguments>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin><plugin><groupId>org.apache.maven.plugins</groupId><artifactId>maven-compiler-plugin</artifactId><configuration><source>8</source><target>8</target></configuration></plugin>
```
Comment (niloc132) [inline jsinterop-ts-defs-processor/pom.xml:131]: fix indentation

File: jsinterop-ts-defs-annotations/pom.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <packaging>jar</packaging>
+
+    <name>tsinterop-annotations</name>
+    <url>https://github.com/Vertispan/jsinterop-ts-defs</url>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.build.outputEncoding>UTF-8</project.build.outputEncoding>
+
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (niloc132) [inline jsinterop-ts-defs-annotations/pom.xml:20]: TODO consider forcing java 11+

File: README.md (hunk 1 above)
Comment (vegegoku) [reply README.md:63]: The idea of the example is to show how the annotation works not to describe a real case.

File: jsinterop-ts-defs-annotations/pom.xml (hunk 4 above)
Comment (vegegoku) [reply jsinterop-ts-defs-annotations/pom.xml:20]: Added a TODO here to check this later

**Pull request merged**